eZ/Bundle/EzPublishCoreBundle/Command/NormalizeImagesPathsCommand.php

```diff
@@ -0,0 +1,161 @@
+{
+    const IMAGE_LIMIT = 100;
+
+    const BEFORE_RUNNING_HINTS = <<<EOT
+<error>Before you continue:</error>
+- Make sure to back up your database.
+- Run this command in production environment using <info>--env=prod</info>
+- Manually clear SPI/HTTP cache after running this command.
+EOT;
+
+    /** @var \eZ\Publish\Core\FieldType\Image\ImageStorage\Gateway\DoctrineStorage */
+    private $imageGateway;
+
+    /** @var \eZ\Publish\Core\IO\FilePathNormalizerInterface */
+    private $filePathNormalizer;
+
+    /** @var \Doctrine\DBAL\Driver\Connection */
+    private $connection;
+
+    public function __construct(
+        DoctrineStorage $imageGateway,
+        FilePathNormalizerInterface $filePathNormalizer,
+        Connection $connection
+    ) {
+        parent::__construct();
+
+        $this->imageGateway = $imageGateway;
+        $this->filePathNormalizer = $filePathNormalizer;
+        $this->connection = $connection;
+    }
+
+    protected function configure()
+    {
+        $beforeRunningHints = self::BEFORE_RUNNING_HINTS;
+
+        $this
+            ->setName('ezplatform:normalize-image-paths')
+            ->setDescription('Normalizes stored paths for images.')
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> normalizes paths for images.
+
+{$beforeRunningHints}
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $io->title('Normalize images paths');
+
+        $io->writeln([
+            'Determining the number of images that require path normalization.',
+            'It may take a minute or two...',
+        ]);
+
+        $imagesCount = $this->imageGateway->countDistinctImages();
+        $imagePathsToNormalize = [];
+        $iterations = ceil($imagesCount / self::IMAGE_LIMIT);
+        $io->progressStart($imagesCount);
+        for ($i = 0; $i < $iterations; ++$i) {
+            $imagesData = $this->imageGateway->getImagesData($i * self::IMAGE_LIMIT, self::IMAGE_LIMIT);
+
+            foreach ($imagesData as $imageData) {
+                $filePath = $imageData['filepath'];
+                $normalizedImagePath = $this->filePathNormalizer->normalizePath($imageData['filepath']);
+                if ($normalizedImagePath !== $filePath) {
+                    $imagePathsToNormalize[] = [
+                        'fieldId' => (int) $imageData['contentobject_attribute_id'],
+                        'oldPath' => $filePath,
+                        'newPath' => $normalizedImagePath,
+                    ];
+                }
+
+                $io->progressAdvance();
+            }
+        }
+        $io->progressFinish();
+
+        $imagePathsToNormalizeCount = \count($imagePathsToNormalize);
+        $io->note(sprintf('Found: %d', $imagePathsToNormalizeCount));
+        if ($imagePathsToNormalizeCount === 0) {
+            $io->success('Nothing to do. Bye!');
```

Comment:
Nice, but not quite our style, I would say :)

```suggestion
            $io->success('No paths to normalize.');
```